src/vsr/superblock.zig

```diff
@@ -812,9 +818,15 @@ pub fn SuperBlockType(comptime Storage: type) type {
             assert((update.storage_size == data_file_size_min) ==
                 update.free_set_reference.empty());
 
+            // NOTE: Within the vsr_state.checkpoint assignment below, do not read from vsr_state
+            // directly. A miscompilation bug (as of Zig 0.11.0) causes fields to receive the
+            // incorrect values.
```

Comment:
(Not sure if this is technically a 'miscompilation' or just underspecification, but it is surprising and worth a note either way.)